Adam Matthew Digital.js

```diff
@@ -142,11 +142,14 @@ function scrapeNew(doc, url) {
 	item.libraryCatalog = extractCatalogName(doc.title);
 	item.url = url.replace('/SearchDetails/', '/Details/')
 		.replace('?SessionExpired=True', '');
-	item.attachments.push({
-		title: "Full Text PDF",
-		mimeType: 'application/pdf',
-		url: attr(doc, 'a[href*="/FullDownload"]', 'href')
-	});
+	let pdfUrl = attr(doc, 'a[href*="/FullDownload"]', 'href');
+	if (pdfUrl) {
+		item.attachments.push({
+			title: "Full Text PDF",
+			mimeType: 'application/pdf',
+			url: pdfUrl
+		});
+	}
```

Comment:
This is actually worth debating. I'm not sure to what extent blindly assigning download URLs and letting them fail if inaccessible vs. detecting their availability ahead of time was a conscious policy decision.

If there's a PDF but you don't have access to it, is it more user-friendly to show it with a red X or not show it at all?

(Curious what @adam3smith thinks of this.)

Reply:
Just to lay out a few obvious arguments:

1. If you get an X, it shows that you're missing something that's available, and you might be able to take steps (e.g., reconnect through your proxy) to remedy that.
1. If you never have access, it's annoying to get what looks like an error all the time.
1. It's sort of nice to show that Zotero could access something if you had access.

Reply:
I don't think we have consistent behavior from a user perspective in translators and it doesn't seem to have ever caused any confusion. Basically, we don't go out of our way to determine whether a user has access to a PDF (which wouldn't always be possible reasonably anyway), but we also don't try to retrieve a PDF where we know it's going to fail (e.g., when we'd have to make up a URL that's only available via DOM). 

I haven't thought this through in detail, but one nice feature of this approach is that the red x is technically meaningful, i.e. it gives us some indication of where things fail (and which debug we'd need) for troubleshooting.